Userland/Services/LookupServer/LookupServer.cpp

```diff
@@ -123,7 +123,8 @@ Vector<DNSAnswer> LookupServer::lookup(const DNSName& name, unsigned short recor
             answer.type(),
             answer.class_code(),
             answer.ttl(),
-            answer.record_data()
+            answer.record_data(),
+            answer.mdns_cache_flush(),
```

Comment:
Actually perhaps we don't want to propagate this value back to our callers and store it into the cache — perhaps we want to handle it at this level (i.e. flush the cache if requested) and drop it. But I'm not exactly sure.

Reply:
Personally I think we should propagate the value. If anything the `DNSAnswer` struct is closer to what we originally received. I don't have a strong opinion on this though.

Reply:
Sounds good, I'll look into that in a bit.

Reply:
I've added code that evicts old cache entries if the mDNS cache flush bit is set.